sessionsManager.js

```diff
@@ -163,21 +171,34 @@ var getSessionByChannelEvent = (messagingEvent) => {
 				apiaiContexts: []
 			}
 
-			userChannelToSessions[messagingEvent.source] = mappedChatSession
-			getChannel(mappedChatSession.channelType).getUserProfile(mappedChatSession.from)
-				.then(json => {
-					console.log("'from' profile:" + JSON.stringify(json))
-					mappedChatSession.profile = json
-					return mappedChatSession
-				})
-				.then(session => {
-					sessionsDb.saveSession(session)
-					return resolve(session)
-				})
-				.catch(error => {
-					console.error("calling get user profile caught an error: " + error)
-					reject(error)
-				})
+			if (messagingEvent.community ) {
+				messagingEvent.community = messagingEvent.community
+			}
+
+			var communityId = (typeof messagingEvent.community != "undefined") ? messagingEvent.community.id : null 
+			tokensDb.getAccessToken(communityId )
+			.then(json => {
+				var access_token = process.env.WORKPLACE_PAGE_ACCESS_TOKEN
+				if (json) {
+					access_token = json.access_token
+					mappedChatSession.communityAccessToken = json.access_token
```

Comment:
I wonder if we should save the session now to store the access_token and avoid future reads to DB